src/components/sockend.js

```diff
@@ -91,24 +91,36 @@ module.exports = class Sockend extends Configurable(Component) {
                 subscribesTo: obj.advertisement.broadcasts,
             }, discoveryOptions);
 
-            subscriber.onMonitorAdded = () => { };
+            subscriber.onMonitorAdded = () => {
+            };
 
             obj.subscriber = subscriber;
 
-            subscriber.on('**', function(data) {
+            subscriber.on('**', function (data) {
                 if (this.event == 'cote:added' || this.event == 'cote:removed') return;
 
                 let topic = this.event.split('::');
                 let namespace = '';
-
                 if (topic.length > 1) {
                     namespace += '/' + topic[0];
                     topic = topic.slice(1);
                 }
-
                 topic = topic.join('');
 
-                io.of(namespace).emit(topic, data);
+                let emitter = io.of(namespace);
+                if (data.__room) {
+                    data.__rooms = new Set(data.__rooms || []);
+                    data.__rooms.add(data.__room);
```

Comment:
do you maybe need to delete `data.__room` here?

Reply:
si senior. will do